torch/custom_class_detail.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+namespace torch {
+namespace jit {
+
+namespace detail {
+
+template <class RetType, class...>
+struct types {
```

Comment:
I looked into this, and I believe its use is not necessary, along with a bunch of heavily templated functions in custom_class.h (addInputs, in particular). We want to minimize template meta-programming locations. Once `    registeredOps.push_back(
        torch::RegisterOperators().op(qualFuncName, std::move(func)));` has been called, the program can simply look up all the information it needs from that operators schema.

Reply:
I've gone through and deleted what I can, but I believe these are still necessary to recover the argument types to instantiate the wrapper lambdas

Reply:
I've just deleted a bunch of template stuff just now, but I believe this in particular is still needed to separate out argument types from the Func type to create the wrapper lambdas in `def_()`